src/control/server/ctl_smd_rpc.go

```diff
@@ -376,101 +376,115 @@ func replaceDevRetryBusy(ctx context.Context, log logging.Logger, e Engine, req
 	return
 }
 
-// SmdManage implements the method defined for the Management Service.
-//
-// Manage SMD devices.
-func (svc *ControlService) SmdManage(ctx context.Context, req *ctlpb.SmdManageReq) (*ctlpb.SmdManageResp, error) {
-	if !svc.harness.isStarted() {
-		return nil, FaultHarnessNotStarted
+func (svc *ControlService) singleDevSmdManage(ctx context.Context, req *ctlpb.SmdManageReq, id string) ([]*ctlpb.SmdManageResp_RankResp, error) {
```

Comment:
Any way to genericize this so there aren't separate functions for single and multi?

Reply:
that's what I had before but I find it cleaner separating them out

Reply:
OK, that's fair.